manifests/group.pp

```diff
@@ -36,6 +36,14 @@
         unless  => "yum grouplist hidden '${name}' | egrep -i '^Installed.+Groups:$'",
         timeout => $timeout,
       }
+      if $ensure == 'latest' {
+        exec { "yum-groupinstall-${name}-latest":
+          command => join(concat(["yum -y groupinstall '${name}'"], $install_options), ' '),
+          onlyif  => "yum groupinfo '${name}' | egrep '\\s+\\+'",
```

Comment:
Can you please explain why you're expecting to filter with `egrep`

Reply:
The logic is looking for `+package` in the list from `groupinfo`. If there is no package with `+` (plus sign) then all packages in the package group are installed and it's not necessary to run `yum groupinstall`. If a package in the package list has `+` then it is in the group but not installed on the system so `yum groupinstall` will ensure all packages get installed.

The existing Exec covers the initial install but the output from `yum grouplist hidden $group` does not change if some packages are added to a group after it's installed.  This PR covers that case.

Reply:
Please comment what the regex is meant to do above the regex.